src/resdk/uploader.py

```diff
@@ -0,0 +1,216 @@
+"""The uploader class.
+
+Supports uploading to S3 bucket and to Genialis server.
+"""
+
+import uuid
+from enum import Enum, auto
+from pathlib import Path
+from typing import TYPE_CHECKING, Dict, Optional, Union
+from urllib.parse import urljoin
+
+import boto3
+import botocore
+
+from resdk.aws.session import get_refreshable_boto3_session
+from resdk.constants import CHUNK_SIZE
+from resdk.exceptions import ResolweServerError
+
+if TYPE_CHECKING:
+    from resdk.resolwe import Resolwe
+
+
+class UploadType(Enum):
+    """Supported uploader types."""
+
+    LOCAL = auto()
+    S3 = auto()
+
+    @staticmethod
+    def default():
+        """Return default upload type."""
+        return UploadType.LOCAL
+
+
+class Uploader:
+    """The uploader class.
+
+    It is used to upload the file to the Genialis platform. Is supports
+    uploading files directly to the server and also directly to the S3 bucket.
```

Comment:
Remove "directly" in front of "to the server".

Reply:
OK.